### ISSUE TITLE
NavigationMeshInstance's Bake NavMesh makes its own enum properties misbehave in Inspector (drop-down list doesn't show)

### ISSUE THREAD

**Issue by EzraT** [godotengine/godot#41899]
**Godot version:**
 v3.2.3.rc6.official

**OS/device including version:**
Linux Mint 20 x86_64

**Issue description:**

Not sure how to describe this issue, but whenever I use the Bake NavMesh button, the editor seems to not show parts of the UI correctly anymore, like drop-down menu's for example.

[video](https://streamable.com/5z7769)

As shown in the video, after clicking the button, the drop-downs do not show up anymore when you click them, not until you select another node and back again, not sure what exactly is happening here.

**Steps to reproduce:**
1. Set up a scene with colliders.
2. Add Navigation nodes. (Navigation, NavigationMeshInstance)
3. Bake NavMesh with the button as shown in the video.
4. Notice drop-downs no longer opening.

**Minimal reproduction project:**
[mrp.zip](https://github.com/godotengine/godot/files/5194354/mrp.zip)

**Comment by mbrlabs** [comment]
Yep, i have the same issue. I can reproduce this 100% of the time i bake a navmesh. Deselecting the node and selecting it again fixes it.

**Comment by smix8** [comment]
I can't recreate this issue at all with Godot 3.3.rc.over9000

@ EzraT @ mbrlabs
Can you still experience this issue on your end with recent Godot 3.3.rc9?

**Comment by EzraT** [comment]
@smix8 
I have not been able to reproduce this anymore in RC9.

**Comment by akien-mga** [comment]
Thanks for testing, closing as fixed.

**Comment by EzraT** [comment]
It seems like this bug has re-surfaced in v3.5.beta1.official

**Comment by akien-mga** [comment]
The video has expired, what parts of the editor UI are not working exactly?

I seem to be able to reproduce it on `3.x` (e17af68daa3e01460c8aa0883c9fd75d139e125b) in the Geometry drop-down properties of NavigationMeshInstance:
![image](https://user-images.githubusercontent.com/4701338/151043893-e672f89e-17f9-42d5-bffd-6011761761a3.png)

Do you also see it in other parts of the editor, or only there?

**Comment by EzraT** [comment]
It seems to only affect dropdown menu's in the editor, they just don't appear for whatever reason.
After trying to click a dropdown menu, it seems like hotkeys do not work anymore either, CTRL+S for example, stops working, as if the dropdown menu is open, its just not visible it seems.

**Comment by akien-mga** [comment]
I made a new recording of the actual issue for clarity:

https://user-images.githubusercontent.com/4701338/151051622-20b72c14-22c8-4549-b8ff-8a9f6b824c0b.mp4

Interesting here the bug shows some inconsistency where the "Parsed Geometry Type" doesn't exhibit the bug (in other tests it did). Clicking `main.tres` to fold the subresource inspector also fails to register on the first click. Reopening it after folding fixes the bug (as re-clicking the NavigationMeshInstance or forcing a reload of the scene would).

**Comment by pycbouh** [comment]
The culprit is the progress dialog that appears as the baking operation is performed. IIRC, it updates by letting a frame pass, which seem to be tripping the inspector somewhere between updates. I've checked, and the controls in the inspector have their correct values inside of them. I've also noticed that pressing the "reset" button next to one of the properties also fixes the inspector, as do the other actions described by others above. That's a good clue that this is the case of "one missing update".

I've checked, and so did @akien-mga, and the appropriate method is called every time any nav mesh property is changed (`_change_notify()`). Which is why I started to think about the progress dialog. I've tried adding a fake update after all the baking and progress showing has been done, and it fixed the issue. Simply adding this to the end of the bake method helped:

```
	p_nav_mesh->set_parsed_geometry_type(p_nav_mesh->get_parsed_geometry_type());
```

So I went about it the other way, and disabled the progress dialog (there are multiple checks in place already in case the object is invalid, so I just commented out the assignment line). And that also fixed the issue. So I think the way the progress dialog goes about updating breaks updates in the inspector. Now we need to find a clean way to trigger one last update in the baking code after all the assignments have been done, because showing progress is good (can't remove it), and the fake assignment that I've used for testing is not good (duh).
